src/error.rs

```diff
@@ -41,20 +41,20 @@ pub enum Error {
     /// underlying connection and you should probably consider them fatal.
     Io(io::Error),
     #[cfg(feature = "tls")]
-    /// TLS error
+    /// TLS error.
     Tls(tls::Error),
     /// - When reading: buffer capacity exhausted.
     /// - When writing: your message is bigger than the configured max message size
     ///   (64MB by default).
-    Capacity(Cow<'static, str>),
+    Capacity(CapacityErrorType),
     /// Protocol violation.
-    Protocol(Cow<'static, str>),
+    Protocol(ProtocolErrorType),
     /// Message send queue full.
     SendQueueFull(Message),
     /// UTF coding error
     Utf8,
     /// Invalid URL.
-    Url(Cow<'static, str>),
+    Url(UrlErrorType),
```

Comment:
Idea: should we call the payload types just `CapacityError`, `ProtocolError`, `UrlError`? The word `Type` inside the names doesn't seem to add a lot of information (the fact that these are enums already says that it is "a type of something").

Reply:
The reason I didn't originally call them just `...Error` was because I hadn't actually implemented the `std::error::Error` trait for each of them. Now that I'm using `thiserror` which automatically implements the error trait, this seems like a good idea.